Dynavity/Dynavity/view/canvas/AnnotationCanvasView.swift

```diff
@@ -3,7 +3,7 @@ import PencilKit
 
 struct AnnotationCanvasView: View {
     @ObservedObject var viewModel: CanvasViewModel
-    @State private var annotationCanvasView = PKCanvasWrapperView()
+    @State private var annotationCanvasView = PKCanvasView()
```

Comment:
Just checking, does this actually need to be a `@State`? `AnnotationCanvasView` itself is essentially a `PKCanvasView` right?

Reply:
Yup, it does. If the `@State` property wrapper were to be removed, then the `AnnotationCanvasView` would not invalidate its appearance and recompute its body when the `annotationCanvasView` is updated. This can be observed by drawing some annotation onto the canvas, switching to the canvas element mode, then zooming/scrolling. The `AnnotationCanvasView` that is responsible for displaying the annotations will not update as the view is not automatically invalidating itself when the `AnnotationCanvasView` receives updates from the `CanvasViewModel`.

Reply:
With regards to your second question, `AnnotationCanvasView` is a wrapper for `PKCanvasView`. Hence, the removal of `PKCanvasWrapperView` as there is no point having two layers of wrappers.

Reply:
From my understanding of how `UIViewRepresentable` works, SwiftUI will call the `makeUIView` method, which returns a `PKCanvasView`. So `AnnotationCanvasView` and `PKCanvasView` are essentially the same?